crates/driver/src/solver/dto/auction.rs

```diff
@@ -1,81 +1,234 @@
+enum Class {
+    Market,
+    Limit,
+    Liquidity,
 }
 
 #[serde_as]
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 struct Token {
     decimals: Option<u8>,
-    alias: Option<String>,
-    external_price: Option<f64>,
-    normalize_priority: Option<u64>,
+    symbol: Option<String>,
     #[serde_as(as = "Option<serialize::U256>")]
-    internal_buffer: Option<U256>,
-    accepted_for_internalization: bool,
+    reference_price: Option<U256>,
+    #[serde_as(as = "serialize::U256")]
+    available_balance: U256,
+    trusted: bool,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "kind", rename_all = "lowercase")]
+enum Liquidity {
+    ConstantProduct(ConstantProductPool),
+    WeightedProduct(WeightedProductPool),
+    Stable(StablePool),
+    ConcentratedLiquidity(ConcentratedLiquidityPool),
+    LimitOrder(ForeignLimitOrder),
 }
 
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ConstantProductPool {
+    id: String,
+    address: H160,
+    #[serde_as(as = "serialize::U256")]
+    gas_estimate: U256,
+    tokens: HashMap<H160, ConstantProductReserve>,
+    #[serde_as(as = "serialize::String")]
+    fee: num::BigRational,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+struct ConstantProductReserve {
+    #[serde_as(as = "serialize::U256")]
+    balance: U256,
+}
+
+#[serde_as]
 #[derive(Debug, Serialize)]
-struct Amm {}
+#[serde(rename_all = "camelCase")]
+struct WeightedProductPool {
+    id: String,
+    address: H160,
+    #[serde_as(as = "serialize::U256")]
+    gas_estimate: U256,
+    tokens: HashMap<H160, WeightedProductReserve>,
+    fee: f64,
+}
 
+#[serde_as]
 #[derive(Debug, Serialize)]
-struct Metadata {
-    environment: Option<String>,
-    auction_id: Option<i64>,
-    run_id: Option<u64>,
-    gas_price: Option<f64>,
-    native_token: Option<H160>,
+struct WeightedProductReserve {
+    #[serde_as(as = "serialize::U256")]
+    balance: U256,
+    weight: f64,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct StablePool {
+    id: String,
+    address: H160,
+    #[serde_as(as = "serialize::U256")]
+    gas_estimate: U256,
+    tokens: HashMap<H160, StableReserve>,
+    amplification_parameter: f64,
+    fee: f64,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct StableReserve {
+    #[serde_as(as = "serialize::U256")]
+    balance: U256,
+    #[serde_as(as = "serialize::U256")]
+    scaling_factor: U256,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ConcentratedLiquidityPool {
+    id: String,
+    address: H160,
+    #[serde_as(as = "serialize::U256")]
+    gas_estimate: U256,
+    tokens: Vec<H160>,
+    #[serde_as(as = "serialize::U256")]
+    sqrt_price: U256,
+    #[serde_as(as = "serialize::U256")]
+    liquidity: U256,
+    tick: i32,
+    #[serde_as(as = "HashMap<_, serialize::U256>")]
+    liquidity_net: HashMap<usize, U256>,
```

Comment:
These are a map of tick -> U256, so this should be:
```suggestion
    #[serde_as(as = "HashMap<serialize::String, serialize::U256>")]
    liquidity_net: HashMap<i32, U256>,
```

Might be worth adding a `Tick` domain type (not needed for the DTO here though).